apps/api/src/app/step-schemas/usecases/generate-preview/generate-preview-use-case.ts

```diff
@@ -0,0 +1,79 @@
+import { Injectable } from '@nestjs/common/decorators';
+import {
+  ControlPreviewIssue,
+  GeneratePreviewResponseDto,
+  GeneratePreviewResponseDtoSchema,
+  TRANSIENT_PREVIEW_PREFIX,
+} from '@novu/shared-internal';
+import { VariableValidatorComponent } from '../../components/variable-validator-component';
+import { GeneratePreviewCommand } from './generate-preview-command';
+import { PreviewStep, PreviewStepCommand } from '../../../bridge/usecases/preview-step';
+import { addKeysToPayloadBasedOnHydrationStrategy } from '../../components/payload-preview-value-generator-component';
+
+function buildResponse(issues: Record<string, ControlPreviewIssue[]>, previewResult, command: GeneratePreviewCommand) {
+  return GeneratePreviewResponseDtoSchema.parse({
+    issues,
+    result: {
+      preview: previewResult.outputs,
+      type: command.stepType,
+    },
+  });
+}
+
+@Injectable()
+export class GeneratePreviewUseCase {
+  constructor(
+    private validator: VariableValidatorComponent,
+    private previewStep: PreviewStep
+  ) {}
+
+  async execute(command: GeneratePreviewCommand): Promise<GeneratePreviewResponseDto> {
+    const issues: Record<string, ControlPreviewIssue[]> = this.validateControlValues(command);
+    const hydratedPayload = addHydrationValuesToPayload(command);
+    const previewResult = await this.executePreviewUsecase(hydratedPayload, command);
+
+    return buildResponse(issues, previewResult, command);
+  }
+
+  private async executePreviewUsecase(hydratedPayload: Record<string, unknown>, command: GeneratePreviewCommand) {
+    const { user } = command;
+    const dto = command.generatePreviewRequestDto;
+    const stepIdInRequest = dto.stepId;
+
+    return await this.previewStep.execute(
+      PreviewStepCommand.create({
+        payload: hydratedPayload,
+        controls: dto.controlValues || {},
+        environmentId: user.environmentId,
+        organizationId: user.organizationId,
+        stepId: stepIdInRequest || TRANSIENT_PREVIEW_PREFIX + command.stepType,
+        userId: user._id,
+        workflowId: dto.workflowId,
```

Comment:
@tatarco we actually need to pass the `NotificationStepEntity.triggers[0].identifier` property to the `PreviewStep` use-case due to the current Worker implementation [depending on the the same property during execution](https://github.com/novuhq/novu/blob/c5047998d5436cf829ebce7840a216b8dd3e42e4/apps/worker/src/app/workflow/usecases/execute-bridge-job/execute-bridge-job.usecase.ts#L108). Given that we have utilize the same Novu-Managed Bridge endpoint for both Preview and Execution, these identifiers MUST be the same.

As a reference, we construct the trigger identifier in the `CreateWorkflow` use-case [here](https://github.com/novuhq/novu/blob/next/libs/application-generic/src/usecases/create-workflow/create-workflow.usecase.ts).

Reply:
Alternatively, we change the API request DTO to specify the slugified `identifier` rather than the MongoID.